crates/uv/tests/it/edit.rs

```diff
@@ -10186,3 +10186,98 @@ fn add_unsupported_git_scheme() {
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     "###);
 }
+
+/// Install a package from an index that requires authentication
+#[test]
+fn add_with_proxy_url() -> anyhow::Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(indoc! {r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = []
+
+        [build-system]
+        requires = ["setuptools>=42"]
+        build-backend = "setuptools.build_meta"
+
+        [[tool.uv.index]]
+        name = "alpha"
+        url = "https://not-real.astral.sh"
+        default = true
+    "#})?;
+
+    uv_snapshot!(context.filters(), context.add()
+        .arg("anyio>=4.3.0")
+        .arg("--index-proxy-url")
+        .arg("alpha=https://public:heron@pypi-proxy.fly.dev/basic-auth/simple"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 4 packages in [TIME]
+    Prepared 4 packages in [TIME]
+    Installed 4 packages in [TIME]
+     + anyio==4.3.0
+     + idna==3.6
+     + project==0.1.0 (from file://[TEMP_DIR]/)
+     + sniffio==1.3.1
+    "
+    );
+
+    let pyproject_toml = context.read("pyproject.toml");
```

Comment:
We should also snapshot the lockfile, right? Maybe that'd be a test case in `crates/uv/tests/it/lock.rs`?